src/bootstrap/build/mod.rs

```diff
@@ -501,6 +501,18 @@ impl Build {
         if target != self.config.build && !target.contains("msvc") {
             base.push(format!("-Clinker={}", self.cc(target).display()));
         }
+        // If we have an external LLVM ensure that crates can link to it
+        let external_llvm = self.config.target_config.get(target).and_then(|config| {
+            config.llvm_config.clone()
+        });
+        if let Some(mut llvm_config) = external_llvm {
+            llvm_config.pop();
+            llvm_config.pop();
+            // Search in <llvm-root>/bin for shared libraries on Windows
+            base.push(format!("-L{}", llvm_config.join("bin").to_str().unwrap()));
+            // Search in <llvm-root>/lib elsewhere
+            base.push(format!("-L{}", llvm_config.join("lib").to_str().unwrap()));
```

Comment:
I would personally like to not land this patch if we're adding linker flags like this, this is not where they're supposed to go. Essentially 100% of the logic of dealing with LLVM should happen as part of `src/librustc_llvm/build.rs`.

Where is the build failing if these aren't specified? This basically means that all external users which do that kind of build will *also* fail which means that we shouldn't require these flags to compile.